lib/blemulator/README.md

```diff
@@ -0,0 +1,253 @@
+blemulator.simulate();
+
+BleManager bleManager = BleManager();
+bleManager.createClient(); //this creates an instance of native BLE
+```
+### Defining simulated peripheral
+The following peripheral is based on [Texas Instruments CC2541 SensorTag](http://www.ti.com/tool/CC2541DK-SENSOR).
+To keep the example clearer, only IR temperature service is simulated.
+```dart
+class SensorTag extends SimulatedPeripheral {
+  SensorTag(
+      {String id = "4C:99:4C:34:DE:76",
+      String name = "SensorTag",
+      String localName = "SensorTag"})
+      : super(
+            name: name,
+            id: id,
+            advertisementInterval: Duration(milliseconds: 800),
+            services: [
+              SimulatedService(
+                  uuid: "F000AA00-0451-4000-B000-000000000000",
+                  isAdvertised: true,
+                  characteristics: [
+                    SimulatedCharacteristic(
+                        uuid: "F000AA01-0451-4000-B000-000000000000",
+                        value: Uint8List.fromList([101, 254, 64, 12]),
+                        convenienceName: "IR Temperature Data"),
+                    SimulatedCharacteristic(
+                        uuid: "F000AA02-0451-4000-B000-000000000000",
+                        value: Uint8List.fromList([0]),
+                        convenienceName: "IR Temperature Config"),
+                    SimulatedCharacteristic(
+                        uuid: "F000AA03-0451-4000-B000-000000000000",
+                        value: Uint8List.fromList([50]),
+                        convenienceName: "IR Temperature Period"),
+                  ],
+                  convenienceName: "Temperature service")
+            ]) {
+    scanInfo.localName = localName;
+  }
+
+  @override
+  Future<bool> onConnectRequest() async {
+    await Future.delayed(Duration(milliseconds: 200));
+    return super.onConnectRequest();
+  }
+}
+```
+
+This creates a peripheral that advertises every 800 milliseconds while peripheral scan is on,
+ waits for 200 milliseconds after receiving connection request before agreeing to connect and has a single service 
+ with three characteristics according to [SensorTag documentation](http://processors.wiki.ti.com/images/a/a8/BLE_SensorTag_GATT_Server.pdf).
+
+The _convenienceName_ fields are optional and not used by the blemulator itself, but allow you to name created objects for better maintainability.
+
+#### Changing advertisement data or scan info
+
+For your convenience you can mark SimulatedService as advertised, but be aware that this doesn't validate the size of advertised services array inside advertisement data.
+
+If you want to have more granular control over what is advertised, simply modify the _scanInfo_ property of the SimulatedPeripheral.
+ The _scanInfo_ is broadcasted automatically while peripheral scan is on each _advertisementInterval_.
+ 
+ScanInfo supports following data:
+```dart
+class ScanInfo {
+  int rssi;
+  int mtu;
```

Comment:
I think it is a mistake by putting `mtu` in `ScanInfo` — MTU is a property of a connection made to peripheral, not scan's property.

Reply:
That's true, it's now removed completely from there.